packages/core/src/agent/Agent.ts

```diff
@@ -214,6 +214,28 @@ export class Agent {
     this._isInitialized = false
   }
 
+  public async rotateKey(walletConfig: WalletConfig) {
```

Comment:
I think this is better located in the `WalletModule` and `Wallet` interface.

Maybe not every wallet handles this the same, so we should probably add a `rotateKey` method to the `Wallet` interface and require to specify the new key and the new key derivation method.